zlp-scheduler/app/controllers/student_controller.rb

```diff
@@ -60,7 +60,7 @@ def update_courses
       subj = Subject.find(params[:dept_id])
       # map to name and id for use in our options_for_select
       @term = Term.find_by active: 1;
-      @courses = Course.where(:abbreviated_subject => subj.subject_code, :term_id => @term.id)
+      @courses = Course.where(:abbreviated_subject => subj.subject_code, :term_id => @term.id).order(course_number: :asc)
```

Comment:
I don't know about ordering this by course number. Maybe by subject would be better so subject stay together?

Reply:
This pulls course numbers for a specific subject, and the function returns only the course numbers. So, I ordered by course number so the list of course numbers would be in order. (This goes to the drop down box in the new schedule form)